src/term/transform/flatten.rs

```diff
@@ -189,10 +189,7 @@ fn make_split_rule(old_rule: &Rule, other_rule: &Rule, def_names: &DefNames) ->
         new_body.subst(other_arg, &new_ctr);
       }
       (Pattern::Num(..), Pattern::Num(..)) => new_pats.push(other_arg.clone()),
-      (Pattern::Num(..), Pattern::Var(..)) => {
-        // How to do this with this kind of number pattern? Subst with a match?
-        todo!();
-      }
+      (Pattern::Num(..), _) => unreachable!(),
```

Comment:
All `(Pattern::Num(..), _) => unreachable!(),` should be unreachable I think, because there can't be patterns nested in num (it's always final, a number). You can remove the line above.